test/jdk/jdk/jfr/event/gc/detailed/PromotionFailedEvent.java

```diff
@@ -49,12 +49,16 @@ public static void test(String testName, String[] vmFlags) throws Throwable {
         // This test can not always trigger the expected event.
         // Test is ok even if no events found.
         List<RecordedEvent> events = RecordingFile.readAllEvents(Paths.get(jfr_file));
+        int minObjectAlignment = 8;
         for (RecordedEvent event : events) {
             System.out.println("Event: " + event);
             long smallestSize = Events.assertField(event, "promotionFailed.smallestSize").atLeast(1L).getValue();
+            Asserts.assertTrue((smallestSize % minObjectAlignment) == 0, "smallestSize " + smallestSize + " is not a valid size.");
```

Comment:
Testing for alignment is a good pragmatic way to check for regressions without adding more logic.

Do the numbers include object headers? If yes, we could assert to >= 8 at least.

Reply:
Yes, the header is included. It is the size we would use if we iterate over a contiguous region the heap. We check that the size is > 0 in the assert above, so it has to be at least 8.